modules/services/gpg-agent.nix

```diff
@@ -237,23 +257,25 @@ in {
           "pinentry-program ${pkgs.pinentry.${cfg.pinentryFlavor}}/bin/pinentry"
           ++ [ cfg.extraConfig ]);
 
-      home.sessionVariablesExtra = optionalString cfg.enableSshSupport ''
-        if [[ -z "$SSH_AUTH_SOCK" ]]; then
-          export SSH_AUTH_SOCK="$(${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket)"
-        fi
-      '';
+      home.sessionVariables = optionalAttrs cfg.enableSshSupport {
+        SSH_AUTH_SOCK = "${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket";
+      };
 
       programs.bash.initExtra = mkIf cfg.enableBashIntegration gpgInitStr;
       programs.zsh.initExtra = mkIf cfg.enableZshIntegration gpgInitStr;
       programs.fish.interactiveShellInit =
         mkIf cfg.enableFishIntegration gpgFishInitStr;
+
+      programs.nushell.extraEnv = mkIf cfg.enableZshIntegration gpgNushellInitStr;
```

Comment:
This condition should probably be `enableNushellIntegration`, not zsh :)

Reply:
oh whoops, good point!